homeassistant/components/gpslogger/device_tracker.py

```diff
@@ -1,39 +1,122 @@
-            gps=gps_location,
-            battery=battery,
-            gps_accuracy=accuracy,
-            attributes=attrs
-        )
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up gpslogger from config entry."""
+    entities = hass.data.setdefault(GPSLOGGER_ENTITIES, {})
+
+    async def async_see(
+            device, latitude, longitude, battery, accuracy, attributes):
+        """Notify the device tracker entities that you see a device."""
+        entity = entities.get(device)
+
+        if entity:
+            await entity.async_seen(
+                latitude=latitude, longitude=longitude,
+                battery=battery, accuracy=accuracy, attributes=attributes)
+            return
+
+        # If no device can be found, create it
+        entity = GPSLoggerEntity(
+            device, latitude, longitude, battery, accuracy, attributes)
+        entities[device] = entity
+        async_add_entities([entity])
 
     hass.data[DATA_KEY] = async_dispatcher_connect(
-        hass, TRACKER_UPDATE, _set_location
+        hass, TRACKER_UPDATE, async_see
     )
     return True
 
 
-async def async_unload_entry(hass: HomeAssistantType, entry):
-    """Unload the config entry and remove the dispatcher connection."""
-    hass.data[DATA_KEY]()
-    return True
+class GPSLoggerEntity(DeviceTrackerEntity):
+    """Represent a tracked device."""
+
+    def __init__(
+            self, device, latitude, longitude, battery, accuracy, attributes):
+        """Set up Geofency entity."""
+        self._accuracy = accuracy
+        self._attributes = attributes
+        self._name = device
+        self._battery = battery
+        self._latitude = latitude
+        self._longitude = longitude
+        self._connected_seen = None
+
+    @property
+    def battery(self):
+        """Return battery value of the device."""
+        return self._battery
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific attributes."""
+        return self._attributes
+
+    @property
+    def gps_accuracy(self):
+        """Return the gps accuracy of the device."""
+        return self._accuracy
+
+    @property
+    def latitude(self):
+        """Return latitude value of the device."""
+        return self._latitude
+
+    @property
+    def longitude(self):
+        """Return longitude value of the device."""
+        return self._longitude
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def source_type(self):
+        """Return the source type, eg gps or router, of the device."""
+        return SOURCE_TYPE_GPS
+
+    async def async_added_to_hass(self):
+        """Register state update callback."""
+        self._connected_seen = self._async_seen
+
+    async def async_will_remove_from_hass(self):
```

Comment:
Also set `self._connected_seen = None`?

Reply:
It doesn't hurt, but the entity will be removed after this is called, so it shouldn't be needed.